In [1]:
import os
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import mannwhitneyu
from scipy import stats

# Model
from sklearn.preprocessing import OneHotEncoder, StandardScaler, QuantileTransformer, MinMaxScaler
from sklearn.impute import SimpleImputer
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline
from sklearn.ensemble import RandomForestRegressor
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import GridSearchCV, train_test_split, RandomizedSearchCV, KFold, RepeatedStratifiedKFold, StratifiedKFold, cross_val_score
from xgboost import XGBRegressor
from sklearn.metrics import root_mean_squared_log_error

# hyperparameter tuning
import optuna

In [33]:
try:
    kaggle_train_file_loc = "../input/playground-series-s4e11/train.csv"
    kaggle_test_file_loc = "../input/playground-series-s4e11/test.csv"
    df_raw = pd.read_csv(kaggle_train_file_loc, index_col=0)
    df_test = pd.read_csv(kaggle_test_file_loc, index_col=0)

except:
    github_train_file_loc = "./dataset/train.csv"
    github_test_file_loc = "./dataset/test.csv"
    df_raw = pd.read_csv(github_train_file_loc, index_col=0, parse_dates=[1])
    df_test = pd.read_csv(github_test_file_loc, index_col=0)

1. How many na values are there? How should i treat na values?
2. What are the data types of each column?
3. Are there any duuplicated values

In [34]:
df = df_raw.copy()
df = df.fillna(0)
df['num_sold'] = df['num_sold'].astype('int')

In [35]:
df.dtypes

date        datetime64[ns]
country             object
store               object
product             object
num_sold             int64
dtype: object

In [36]:
df.isna().sum()

date        0
country     0
store       0
product     0
num_sold    0
dtype: int64

In [32]:
df

id
0            0
1          973
2          906
3          423
4          491
          ... 
230125     466
230126    2907
230127    2299
230128    1242
230129    1622
Name: num_sold, Length: 230130, dtype: int64

# Appendix

### A1: Should I treat the nulls as zeroes in this dataset? What is/are the justification(s) for me to do so?

In [44]:
df_raw.query("num_sold == 0").shape[0]

0

There are 

In [43]:
df_raw["num_sold"].isnull().sum()

8871

There are 8871 null values in the `num_sold` column of the dataset

In [41]:
df_raw["num_sold"].apply(lambda x: x.is_integer()).sum()

221259

In [22]:
df.shape[0] - 221259

8871

### Next